### PR TITLE
Refactor dash API into two separate requests

### PR DESCRIPTION
**Pull request by samuelli**
This change splits incoming and outgoing pull requests into two separate requests. This allows outgoing pull requests to be displayed much faster than outgoing pull requests.

### CODE REVIEW

File: src/client/public/scripts/dash/dash.ts [hunk 1]
```diff
@@ -23,56 +25,56 @@ type StatusDisplay = {
...
+  render(
       prListTemplate(
-          data.incomingPrs,
+          data.prs,
           newlyActionablePRs,
-          '🎉 No incoming pull requests! When you\'re added as a reviewer to a pull request, it\'ll appear here.')}
-  </div>
-  `;
-};
+          'You have no outgoing pull requests. When you open new pull requests, they\'ll appear here'),
```
Comment (gauntface) [inline src/client/public/scripts/dash/dash.ts]: Optional: Moving this to a constant would make parsing the code a little easier. (Same for PR below).

File: src/client/public/scripts/dash/dash.ts [hunk 2]
```diff
@@ -253,67 +255,70 @@ function prTemplate(pr: api.PullRequest, newlyActionablePRs: string[]) {
...
-  lastPolledData = newData;
+  // Send fetches in parallel.
+  const outgoing =
+      fetch(`/api/dash/outgoing${loginParams}`, {credentials: 'include'});
+  const incoming =
+      fetch(`/api/dash/incoming${loginParams}`, {credentials: 'include'});
+
+  // Render outgoing first.
+  const outgoingData =
+      await (await outgoing).json() as api.OutgoingDashResponse;
```
Comment (gauntface) [inline src/client/public/scripts/dash/dash.ts]: Optional: Both incoming and outgoing look like that can be done in parallel 

```javascript
await Promise.all([
  getAndRenderOutgoing(),
  getAndRenderIncoming(),
]);
```
Main reasoning is it should clean up the outgoing and and incoming awaits.

File: src/server/apis/dash-data.ts [hunk 3]
```diff
@@ -2,254 +2,311 @@ import * as express from 'express';
...
+      loginDetails,
+      req.query.login || loginDetails.username,
+      loginDetails.githubToken,
+  );
 
-        if (pr.viewerSubscription === 'IGNORED') {
-          continue;
-        }
+  res.header('content-type', 'application/json');
+  res.send(JSON.stringify(userData, null, 2));
```
Comment (gauntface) [inline src/server/apis/dash-data.ts]: You can use `res.json(userData)` to set header and post.

File: src/client/public/scripts/dash/dash.ts (hunk 2 above)
Comment (samuelli) [reply src/client/public/scripts/dash/dash.ts]: Done.

File: src/client/public/scripts/dash/dash.ts (hunk 1 above)
Comment (samuelli) [reply src/client/public/scripts/dash/dash.ts]: Done.

**Pull request merged**